internal/promqle2etest/backend_otelgcm.go

```diff
@@ -0,0 +1,225 @@
+    metric_statements:
+    - context: datapoint
+      statements:
+      - set(attributes["exported_location"], attributes["location"])
+      - delete_key(attributes, "location")
+      - set(attributes["exported_cluster"], attributes["cluster"])
+      - delete_key(attributes, "cluster")
+      - set(attributes["exported_namespace"], attributes["namespace"])
+      - delete_key(attributes, "namespace")
+      - set(attributes["exported_job"], attributes["job"])
+      - delete_key(attributes, "job")
+      - set(attributes["exported_instance"], attributes["instance"])
+      - delete_key(attributes, "instance")
+      - set(attributes["exported_project_id"], attributes["project_id"])
+      - delete_key(attributes, "project_id")
+
+  batch:
+    # batch metrics before sending to reduce API usage
+    send_batch_max_size: 200
+    send_batch_size: 200
+    timeout: 5s
+
+  memory_limiter:
+    # drop metrics if memory usage gets too high
+    check_interval: 1s
+    limit_percentage: 65
+    spike_limit_percentage: 20
+  %s
+
+# Note that the googlemanagedprometheus exporter block is intentionally blank
+exporters:
+  googlemanagedprometheus:
+    project: "%s"
+    # TODO(bwplotka): Do we need to disable cumulative_normalization explicitly?
+    metric:
+      # TODO(bwplotka): Update docs? Change defaults? Those metrics are a bad idea to include (IMO).
+      # We are working in Prometheus OSS to slowly promote more resource attr as labels (if not all one day).
+      extra_metrics_config:
+        enable_target_info: false
+        enable_scope_info: false
+
+service:
+  pipelines:
+    metrics:
+      receivers: [prometheus]
+      processors: [batch, memory_limiter, resource, transform%s]
+      exporters: [googlemanagedprometheus]
+  telemetry:
+    # Internal collector metrics for debugging.
+    # https://opentelemetry.io/docs/collector/internal-telemetry/#configure-internal-metrics
+    metrics:
+      readers:
+        - pull:
+            exporter:
+              prometheus:
+                host: '0.0.0.0'
+                port: 9090
+`, scrapeTargetAddress, name, cluster, location, extraProcessorsConfigs, project, extraProcessorsEnable)
+	if err := os.WriteFile(filepath.Join(f.Dir(), "config.yml"), []byte(config), 0600); err != nil {
+		return e2emon.AsInstrumented(e2e.NewFailedRunnable(name, fmt.Errorf("create otel config failed: %w", err)), "http")
+	}
+
+	if err := os.WriteFile(filepath.Join(f.Dir(), "gcm-sa.json"), gcmSA, 0600); err != nil {
+		return e2emon.AsInstrumented(e2e.NewFailedRunnable(name, fmt.Errorf("write gcm SA failed: %w", err)), "http")
+	}
+
+	args := map[string]string{
+		"--config": filepath.Join(f.Dir(), "config.yml"),
+		"--feature-gates=receiver.prometheusreceiver.EnableNativeHistograms": "",
```

Comment:
did you need to enable the `receiver.prometheusreceiver.RemoveStartTimeAdjustment` feature gate as well?